# HG changeset patch
# User Jaroslav Hajek <highegg@gmail.com>
# Date 1228988393 -3600
#      Thu Dec 11 10:39:53 2008 +0100
# Node ID 4780279e809445e91a7cab8218a622ffa70d81cc
# Parent  b65c75203cef376295a5fe3450ee50042d5adeb1
fix dm-dm binary ops

diff --git a/liboctave/ChangeLog b/liboctave/ChangeLog
--- a/liboctave/ChangeLog
+++ b/liboctave/ChangeLog
@@ -1,8 +1,12 @@
+2008-12-11  Jaroslav Hajek  <highegg@gmail.com>
+
+	* mx-op-defs.h (DMDM_BIN_OP): Fix invalid buffer length.
+
 2008-12-10  Jaroslav Hajek  <highegg@gmail.com>
 
 	* dMatrix.h, dMatrix.cc (Matrix::expm): Remove.
 	* fMatrix.h, fMatrix.cc (FloatMatrix::expm): Remove.
 	* CMatrix.h, CMatrix.cc (ComplexMatrix::expm): Remove.
 	* fCMatrix.h, fCMatrix.cc (FloatComplexMatrix::expm): Remove.
 
 2008-12-09  Jaroslav Hajek  <highegg@gmail.com>
diff --git a/liboctave/mx-op-defs.h b/liboctave/mx-op-defs.h
--- a/liboctave/mx-op-defs.h
+++ b/liboctave/mx-op-defs.h
@@ -1182,17 +1182,17 @@ operator * (const DM& dm, const M& m) \
     if (dm1_nr != dm2_nr || dm1_nc != dm2_nc) \
       gripe_nonconformant (#OP, dm1_nr, dm1_nc, dm2_nr, dm2_nc); \
     else \
       { \
 	r.resize (dm1_nr, dm1_nc); \
  \
 	if (dm1_nr > 0 && dm1_nc > 0) \
 	  F ## _vv (r.fortran_vec (), dm1.data (), dm2.data (), \
-		    dm1_nr * dm2_nc); \
+		    dm1.length ()); \
       } \
  \
     return r; \
   }
 
 #define DMDM_BIN_OPS(R, DM1, DM2) \
   DMDM_BIN_OP (R, operator +, DM1, DM2, mx_inline_add) \
   DMDM_BIN_OP (R, operator -, DM1, DM2, mx_inline_subtract) \
